src/lightning_app/components/database/server.py

```diff
@@ -172,6 +182,15 @@ def db_url(self) -> Optional[str]:
             return f"http://{self.internal_ip}:{self.port}"
         return self.internal_ip
 
-    def on_exit(self):
-        self.drive.put(self.db_filename)
+    def periodic_store_database(self, store_interval):
+        while True:
+            self.store_database()
+            time.sleep(store_interval)
+
+    def store_database(self):
+        with self._store_lock:
```

Comment:
The lock should also be used within the FastAPI request handler to avoid updating the database while copying it.

Reply:
This prompted to get more in depth. You should be able to copy a SQLite file at any time while it's being updated, but due to a Unix quirckiness, it can lead to issues when done from a thread on POSIX systems.

I'm going to update this PR so that we make a proper backup with the official API and then sync.